prov/efa/src/rdm/rxr_ep.c

```diff
@@ -996,6 +1059,37 @@ static int efa_set_fi_hmem_p2p_opt(struct rxr_ep *rxr_ep, int opt)
 	return -FI_EINVAL;
 }
 
+/**
+ * @brief set cuda_api_permitted flag in rxr_ep
+ * @param[in,out]	ep			endpoint
+ * @param[in]		cuda_api_permitted	whether cuda api is permitted
+ * @return		0 on success,
+ *			-FI_EOPNOTSUPP if endpoint relies on CUDA API call to support CUDA memory
+ * @related rxr_ep
+ */
+static int rxr_ep_set_cuda_api_permitted(struct rxr_ep *ep, bool cuda_api_permitted)
+{
+	if (!hmem_ops[FI_HMEM_CUDA].initialized) {
+		EFA_WARN(FI_LOG_EP_CTRL, "FI_OPT_CUDA_API_PERMITTED cannot be set when "
+			 "CUDA library or CUDA device is not available");
+		return -FI_EINVAL;
```

Comment:
Same comment as above. Is there a special reason to make this option restrictive and tied to GPU availability?

Reply:
If there is no CUDA library or CUDA device, then this option does not apply.